framework/src/util/PropsCheck.cpp

```diff
@@ -0,0 +1,75 @@
+/*=============================================================================
+
+ Library: CppMicroServices
+
+ Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+ file at the top-level directory of this distribution and at
+ https://github.com/CppMicroServices/CppMicroServices/COPYRIGHT .
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+
+ =============================================================================*/
+
+#include "PropsCheck.h"
+
+namespace cppmicroservices {
+namespace props_check {
+namespace {
+const Any emptyAny;
+}
+
+/**
+ * @brief Validates that the provided AnyMap conforms to the same constraints that
+ * those stored in Property objects have.
+ * 
+ * The provided AnyMap is said to be valid if there exists no pairs of two keys
+ * which differ in case only (e.g., "service.feature", "Service.feature"). If this
+ * condition is not true, this function throws as defined below.
+ * 
+ * @param am The AnyMap to validate
+ * @throws std::runtime_error Thrown when `am` is invalid (described above)
+ */
+void ValidateAnyMap(const cppmicroservices::AnyMap& am)
```

Comment:
Check to see if using a std::unordered_set (case-insensitive) is a better solution for checking validity.

Reply:
It's not.